manifests/argocd-operator.yaml

```diff
@@ -0,0 +1,3196 @@
+              type: string
+            phase:
+              description: 'Phase is a simple, high-level summary of where the ArgoCD is in its lifecycle. There are five possible phase values: Pending: The ArgoCD has been accepted by the Kubernetes system, but one or more of the required resources have not been created. Available: All of the resources for the ArgoCD are ready. Failed: At least one resource has experienced a failure. Unknown: For some reason the state of the ArgoCD phase could not be obtained.'
+              type: string
+            redis:
+              description: 'Redis is a simple, high-level summary of where the Argo CD Redis component is in its lifecycle. There are five possible redis values: Pending: The Argo CD Redis component has been accepted by the Kubernetes system, but one or more of the required resources have not been created. Running: All of the required Pods for the Argo CD Redis component are in a Ready state. Failed: At least one of the  Argo CD Redis component Pods had a failure. Unknown: For some reason the state of the Argo CD Redis component could not be obtained.'
+              type: string
+            repo:
+              description: 'Repo is a simple, high-level summary of where the Argo CD Repo component is in its lifecycle. There are five possible repo values: Pending: The Argo CD Repo component has been accepted by the Kubernetes system, but one or more of the required resources have not been created. Running: All of the required Pods for the Argo CD Repo component are in a Ready state. Failed: At least one of the  Argo CD Repo component Pods had a failure. Unknown: For some reason the state of the Argo CD Repo component could not be obtained.'
+              type: string
+            server:
+              description: 'Server is a simple, high-level summary of where the Argo CD server component is in its lifecycle. There are five possible server values: Pending: The Argo CD server component has been accepted by the Kubernetes system, but one or more of the required resources have not been created. Running: All of the required Pods for the Argo CD server component are in a Ready state. Failed: At least one of the  Argo CD server component Pods had a failure. Unknown: For some reason the state of the Argo CD server component could not be obtained.'
+              type: string
+          type: object
+      type: object
+  version: v1alpha1
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: true
+---
+apiVersion: v1
+kind: ServiceAccount
```

Comment:
Are these argocd rbac objects not created dynamically by the operator based on the namespace the ArgoCD CRD is found in?

Reply:
This is a really good catch. Looks like they did that in the latest tag (v0.0.14): https://github.com/argoproj-labs/argocd-operator/blob/v0.0.14/deploy/role.yaml but they fixed this (remove all rbac objects of ArgoCD Cluster from the operator) in master branch: https://github.com/argoproj-labs/argocd-operator/blob/master/deploy/role.yaml

I will generate this file again from `master`. Then will separate CRDs to CRDs phase as well.

Reply:
Looks like removing the RBAC object break the funationality of the operator + argocd cluster. Seems like a bug from upstream. I will dig into this tomorrow.

Ref: https://github.com/argoproj-labs/argocd-operator/pull/224

Reply:
So basically, they remove that ArgoCD Cluster's RBCA objects from manifest and make the operator create them in the above PR but they haven't released a version with that change yet. We might need to wait for them to release a new version that includes that fix.

Reply:
I think lets at least split the RBAC for the operator vs the RBAC for the clusters that come up into separate files

Reply:
I have removed the RBAC of the cluster from installing phase and split them into a separate file and in the test phase, I use that file to deploy necessary RBAC for the cluster it can come up healthy.